File: backend/src/main/java/de/gregoryseibert/baeckereibackend/model/NewsItem.java
package de.gregoryseibert.baeckereibackend.model;

import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;

/**
 * The NewsItem class represents a news item.
 *
 * @author Gregory Seibert
 * @version 1.0
 */

@Data
@Entity
public class NewsItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotEmpty
    private String title;

    @Lob
    private String content;

    private String pictureFilename;
}

File: frontend-app/Baeckerei/model/NewsItem.swift
//
//  NewsItem.swift
//  Baeckerei
//
//  Created by Gregory Seibert on 10.02.19.
//  Copyright © 2019 Gregory Seibert. All rights reserved.
//

import Foundation

class NewsItem: Codable {
    let id: Int
    var title: String
    var content: String
    var pictureFilename: String
}
